mysqld_exporter.go

```diff
@@ -104,8 +103,7 @@ var scrapers = map[collector.Scraper]bool{
 	collector.ScrapeReplicaHost{}:                         false,
 }
 
-func parseMycnf(config interface{}) (string, error) {
-	var dsn string
+func parseMycnf(config interface{}) (dsn string, err error) {
```

Comment:
typically we try and stay away from named returns. I would remove the names here.